contrib/natss/pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -73,8 +81,25 @@ func createReceiverFunction(s *SubscriptionsSupervisor, logger *zap.SugaredLogge
 			logger.Errorf("Error during marshaling of the message: %v", err)
 			return err
 		}
-		if err := stanutil.Publish(s.natssConn, ch, &message, logger); err != nil {
+		s.natssConnMux.Lock()
+		currentNatssConn := s.natssConn
+		s.natssConnMux.Unlock()
+		if currentNatssConn == nil {
+			return fmt.Errorf("No Connection to NATSS")
+		}
+		if reflect.ValueOf(currentNatssConn).IsNil() {
```

Comment:
I'm not familiar with Go reflection, how does this differ from `if currentNatssConn == nil` three lines above?

Reply:
this check helps to prevent panic when we do (*currentNattsConn).NatssCon().IsConnected(), as during debugging I cold see currentNattsConn is not nil but it point to nil interface when the connection gets lost. reflect.ValueOf allows us safely evaluate that the interface is not nil. Let me know if it makes sense.